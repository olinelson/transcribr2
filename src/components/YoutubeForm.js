import React from 'react'

import Link from 'gatsby'

import { uploadYoutube } from '../services/clipManagement'
import { LinkOutlined } from '@ant-design/icons'

import { Form, Input, Button } from 'antd'

function YoutubeForm ({ appState, setAppState }) {
  const onFinish = async ({ url }) => {
    uploadYoutube({ appState, url, setAppState })
  }

  return (
    <Form
      name='youtube-form'
      onFinish={onFinish}
    >

      <Form.Item
        name='url'
        rules={[{ required: true, message: 'Please input a youtube url!' }]}
      >
        <Input prefix={<LinkOutlined className='site-form-item-icon' />} placeholder='https://www.youtube.com/watch?v=376q9QUezJU' />
      </Form.Item>

      <Form.Item>
        <Button loading={appState.youtubeUploading} type='primary' htmlType='submit' className='login-form-button'>
          Add
        </Button>
      </Form.Item>
    </Form>

  )
}

export default YoutubeForm
