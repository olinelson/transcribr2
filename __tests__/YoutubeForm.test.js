import React from 'react'
import renderer from 'react-test-renderer'

import YoutubeForm from '../src/components/YoutubeForm'

beforeAll(() => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn()
    }))
  })
})

const appState = {
  user: {},
  clips: [],
  uploadDrawerOpen: false,
  uploadYoutubeDrawerOpen: false,
  searchClipDrawerOpen: false,
  editUserDrawerOpen: false,
  editClipDrawerOpen: false,
  editWordDrawerOpen: false,
  editEmailDrawerOpen: false,
  youtubeUploading: false,
  offline: false
}

describe('YoutubeForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<YoutubeForm appState={appState} setAppState={() => null} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
