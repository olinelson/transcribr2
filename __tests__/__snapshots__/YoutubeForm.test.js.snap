// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubeForm renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="youtube-form"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <span
            className="ant-input-affix-wrapper"
            onMouseUp={[Function]}
          >
            <span
              className="ant-input-prefix"
            >
              <span
                aria-label="link"
                className="anticon anticon-link site-form-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="link"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M574 665.4a8.03 8.03 0 00-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 00-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 000 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 000 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 00-11.3 0L372.3 598.7a8.03 8.03 0 000 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                  />
                </svg>
              </span>
            </span>
            <input
              className="ant-input"
              id="youtube-form_url"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="https://www.youtube.com/watch?v=376q9QUezJU"
              style={null}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn login-form-button ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Add
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
